tools/coqdep/lib/dune

```diff
@@ -3,3 +3,13 @@
  (libraries coq-core.boot coq-core.lib findlib.internal))
 
 (ocamllex lexer)
+
+(rule
+ (targets static_toplevel_libs.ml)
+ (deps %{project_root}/_build/install/%{context_name}/lib/coq-core/META)
+ (action
+  (with-stdout-to %{targets}
+   (run ocamlfind query -recursive -predicates native coq-core.toplevel
+        -prefix "let static_toplevel_libs = [\n"
+        -format "\"%p\";"
+        -suffix "\n]\n"))))
```

Comment:
@ejgallego I ended up writing this rule instead of hardcoding the list

Reply:
It's actually `tools/coqdep/lib/static_toplevel_libs.ml` but with that change it works fine

Reply:
Oh, somehow I missed this line in the diff
```
 (deps %{project_root}/_build/install/%{context_name}/lib/coq-core/META)
```
so indeed things are fine.

This is unfortunately not officially supported dune upstream, I think. I guess we can keep it.

Maybe we should open a Dune issue to add a `%{meta:pkg}` variable, the patch should be easy as I already wrote the OCaml code to resolve the META in dune.

Reply:
Internally, we've been turning a few `%{project_root}` into `%{workspace_root}` to support workspace builds (the `_build` directory is always at the root of the workspace, not the root of the project). Is there a good reason to prefer `%{project_root}` here?

Reply:
the path should probably be the same as https://github.com/coq/coq/blob/0e0f4abfa4df8ee37d4b5782ea997c0cc434730b/dune#L44
if you want to make a PR to change that one feel free

Reply:
Indeed, we should use `%{workspace_root}` cc @SkySkimmer 

@rlepigre , can you share your use case for depending on `%{workspace_root}`, is also due to `META` ? Or you just prefer to use absolute paths in `(dep ...)`

Reply:
@SkySkimmer I've checked and using `%{project_root}/_build` is a bug, it must be fixed both in the current tree and in this PR. That was a good catch, thanks @rlepigre .

Reply:
@ejgallego how about merging this PR now and then you can fix the path in https://github.com/coq/coq/pull/19899 ?

Reply:
As you prefer, that works for me.

Reply:
Yeah, no specific use-case, that's just a bug indeed.